pkg/upgrade/key_pair_test.go

```diff
@@ -0,0 +1,187 @@
+package upgrade_test
+
+import (
+	"errors"
+	"fmt"
+	"testing"
+
+	"github.com/vmware/cluster-api-upgrade-tool/pkg/upgrade"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"sigs.k8s.io/cluster-api/pkg/apis/cluster/v1alpha1"
+)
+
+type secrets struct {
+	secret *v1.Secret
+	err    error
+}
+
+func (s *secrets) Get(_ string, _ metav1.GetOptions) (*v1.Secret, error) {
+	return s.secret, s.err
+}
+
+func TestNewRestConfigFromCASecretRef(t *testing.T) {
+	secret := &secrets{
+		secret: &v1.Secret{
+			Data: map[string][]byte{
+				"cert": []byte(caCertificate),
+				"key":  []byte(caPrivateKey),
+			},
+		},
+	}
+	config, err := upgrade.NewRestConfigFromCASecretRef(secret, "name", "clustername", "https://example.com:6443")
+	if err != nil {
```

Comment:
Feel free to pull in testify and do `require.NoError(t, err)`, etc

Reply:
that seems like a big dependency for a very small syntax change. Is there any other reason you think we should be pulling that in?

Reply:
I quite like testify for unit testing. If we pull it in, we can use it in all our tests.

Reply:
I don't think we need it for this statement as `err != nil` is an extremely common pattern in go. If it provides value in some other way in the future I'll be happy to pull it in

Reply:
Fair enough